### DISCUSSION TITLE
局域网地址配置hosts不成功

### DISCUSSION THREAD

**Discussion by ron159** [IrineSistiana/mosdns#847] [Q&A]
Hi,各位吴彦祖
我有一个需求，通过hosts将局域网内将某个域名A映射到某一局域网IP（这个域名在外部同样被解析到本地的公网IP上），这样可以实现内外网切换的时候，对应的app无缝访问；之前这样配置一直是正常工作的，最近我重装了openwrt，一切按以前一样配置，但是现在无论如何在内网都无法解析这个地址了；
用wireshark抓包看，mosdns实际对dns请求进行了响应，但是返回的repsonse没有ansser（与此同时，我在hosts里面配置的其他外网IP，都可以正常查询到）
![PixPin_2024-09-29_23-02-35](https://github.com/user-attachments/assets/5b3f5bcc-cac4-4616-bf80-32aa5b0d6b06)
![PixPin_2024-09-29_23-05-10](https://github.com/user-attachments/assets/de0ee6ac-5f65-4cd5-bba2-0cf3d08dfc98)

![PixPin_2024-09-29_23-04-07](https://github.com/user-attachments/assets/94b8bda4-1006-48ec-b301-731ea4f1e3cb)

mosdns的日志里面没有打印异常的信息，所以暂时没有附上，不知道大家有没有什么排查思想，谢谢

**Comment by sam452167735789** [comment]
你要去dnsmasq里把屏蔽.lan的选项框里的lan删除。
